### ISSUE TITLE
Dit is wel next level shit!!

### ISSUE THREAD

**Issue by olijf** [kjhermans/meaningful#1]
I'm impressed.

Would it be relevant to compare this to Google's protobuf?

**Comment by kjhermans** [comment]
Yes and no. No, because in the current implementation, string encoding
overhead is a bit high (12.5% - however, this is the standard overhead of
the L2 on a serial line and nobody complains about that). Yes, because it
plays nicer with really, really small buffers and lines that may lose part
of the tail end of your message. Also: there are simple ways to reduce that
overhead. And of course also: correctness. A parser that cannot fail is a
parser that is always correct.
